test/update.js

```diff
@@ -27,3 +27,75 @@ test('models returned by update are emitted by the model stream', function (t) {
     t.end()
   }))
 })
+
+test('model stream will not emit a model if is a duplicate', function (t) {
+  t.plan(1)
+  var initialModel = {initial: true}
+  var app = {
+    init: function () {
+      return {model: initialModel}
+    },
+    update: function (model, action) {
+      return {model: model}
+    },
+    view: function (model, dispatch) {
+      dispatch()
+      return inu.html`<div></div>`
+    }
+  }
+  var sources = inu.start(app)
+
+  pull(sources.models(), pull.drain(function (model) {
+    t.equal(model, initialModel)
+  }))
+})
+
+test('effects stream will not emit an effect if update returns a null effect', function (t) {
+  t.plan(1)
+  var initialModel = {initial: true}
+  var app = {
+    init: function () {
+      return {model: initialModel}
+    },
+    update: function (model, action) {
+      t.ok(model)
+      return {model, effect: null}
+    },
+    view: function (model, dispatch) {
+      dispatch()
+    },
+    run: function (effect) {
+      t.error(effect)
```

Comment:
[i reckon we should use `t.fail` here](https://github.com/substack/tape#tfailmsg)